### PR TITLE
multi line layout issue with white mod_articles_category

### LINKED ISSUE #12598: mod_articles_category - default.php - multi line - layout failure
screen shot 2016-10-27 at 12 17 55

if title is longer than one line
after deleting all tab and linebreak between [a] and [/a] in source code it works correctly

### PR DESCRIPTION
**Pull request by PhilETaylor**
fixes #12598 @TorstenE to provide further details for testing

Pull Request for Issue #12598 .

### Summary of Changes

Remove Whitespace within link a tag to prevent line wrapping

### CODE REVIEW

File: modules/mod_articles_category/tmpl/default.php [hunk 1]
```diff
@@ -83,9 +83,7 @@
 		<?php foreach ($list as $item) : ?>
 			<li>
 				<?php if ($params->get('link_titles') == 1) : ?>
-					<a class="mod-articles-category-title <?php echo $item->active; ?>" href="<?php echo $item->link; ?>">
-						<?php echo $item->title; ?>
-					</a>
+					<a class="mod-articles-category-title <?php echo $item->active; ?>" href="<?php echo $item->link; ?>"><?php echo $item->title; ?></a>
```
Comment (n9iels) [inline modules/mod_articles_category/tmpl/default.php]: Joomla! has a build in JHtml function to do this. You can replace this line with:
```php
<?php echo JHtml::link($item->link, $item->title, array("class" => "mod-articles-category-title " . $item->active)); ?>
```

**Comment by n9iels** [comment]
Hi @PhilETaylor,

Thanks for your contribution! I placed a comment on your changes. Can you please do the same for these lines:
https://github.com/PhilETaylor/joomla-cms/blob/a54e1e724385b65e9ae5d40a707f72df4e779dc7/modules/mod_articles_category/tmpl/default.php#L22-L24

File: modules/mod_articles_category/tmpl/default.php (hunk 1 above)
Comment (mbabker) [inline modules/mod_articles_category/tmpl/default.php]: If the use of this function is going to be standardized in the CMS, it should be done as a separate PR.  This is fine as is.

**Comment by PhilETaylor** [comment]
> If the use of this function is going to be standardized in the CMS, it should be done as a separate PR. This is fine as is.

So I'll not make any further changes to this PR then. 

Lets hope @TorstenE comes back and tests if it fixes his issue,

**Comment by sanderpotjer** [comment]
I have tested this item :white_check_mark: successfully on a54e1e724385b65e9ae5d40a707f72df4e779dc7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13561">issues.joomla.org/tracker/joomla-cms/13561</a>.</sub>

**Comment by marrouchi** [comment]
Unable to reproduce the issue ... :(

File: modules/mod_articles_category/tmpl/default.php (hunk 1 above)
Comment (PhilETaylor) [reply modules/mod_articles_category/tmpl/default.php:86]: Please close this review for completeness...

**Comment by PhilETaylor** [comment]
Paging... @TorstenE Please come back and test if this fixes your issue,

File: modules/mod_articles_category/tmpl/default.php (hunk 1 above)
Comment (brianteeman) [inline modules/mod_articles_category/tmpl/default.php]: @PhilETaylor there is no way to do that :(

**Comment by Quy** [comment]
@TorstenE has not been active on GitHub since reporting this issue. I doubt that you will hear from him soon. I am wondering if this is a CSS issue on his end.

**Comment by TorstenE** [comment]
I had it changed in my own files myself. Yes, it works wonderfully - thank you

**Comment by PhilETaylor** [comment]
Excellent - this has had two good tests, please RTC (maybe @franz-wohlkoenig can update the test of @TorstenE on issues.joomla.org to reflect the successful test. 

> @Quy I am wondering if this is a CSS issue on his end.

Nope, a known html issue that can cause issues

**Comment by TorstenE** [comment]
I have tested this item :white_check_mark: successfully on a54e1e724385b65e9ae5d40a707f72df4e779dc7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13561">issues.joomla.org/tracker/joomla-cms/13561</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.
